_posts/2023-05-25-Announcing-Data-Prepper-2.3.0.md

```diff
@@ -0,0 +1,147 @@
+
+### Setting event metadata keys
+
+`add_entries` processor can also add entries in the event’s metadata instead of the event itself. And example `add_entries` processor configuration for adding an entry to metadata 
+
+```
+processor:
+   - add_entries:
+       entries:
+         - metadata_key: "request_len"
+           value_expression: "length(/request)"
+```
+This configuration adds an attribute to event metadata with attribute key `request_len` with value equal to the length of the request field in the key. The value can be set using value or format or `value_expression` option of entries field.
+
+
+## S3 sink
+
+Data Prepper now supports saving data to Amazon S3 sinks as ndjson. Amazon S3 is a popular choice for storing large volumes of data reliably and in a cost-effective way.
+
+Ingesting data into S3 offers a lot of possibilities for your data pipelines, including some of the following:
+
+* Noisy or uninteresting data can be routed to S3 and not to OpenSearch to reduce load on your OpenSearch cluster. This can also help you save on compute and storage costs.
+* Ingesting data to S3 creates data which you can use for future processing.
+
+
+## Tail sampling
+
+Data Prepper 2.3.0 supports tail sampling to limit the number of events that are sent to a sink similar to the tail sampling support provided by open telemetry. More details on tail sampling in [OpenTelemetry](https://opentelemetry.io) can be found [here](https://opentelemetry.io/blog/2022/tail-sampling/).
+
+Tail processing in Data Prepper is supported as an action to aggregate processor. The events are stored in the aggregate processor beyond the `group_duration` time until no events are received in the last `wait_period` time. 
+
+For example, the following configuration sends all traces with errors to the sink and non-error events are sampled with the user specified `percent` probabilistic sampler. 
+
+```
+trace-normal-pipeline:
+  source:
+    otel_trace_source:
+      ssl: false
+  processor:
+    - trace_peer_forwarder:
+  processor:
+    - aggregate:
+        identification_keys: ["traceId"]
+        action:
+          tail_sampler:
+            percent: 20
+            wait_period: "15s"
+            error_condition: "/traceGroupFields/statusCode == 2"
+        group_duration: "30s"
+ sink:
+    - opensearch
+         hosts: ["https://opensearch:9200"]
+         insecure: true
+         username: "admin"
+         password: "admin"
+         index: sampled-traces
+
+```
+
+## Getting started
+
+* To download Data Prepper, see the [OpenSearch downloads](https://opensearch.org/downloads.html) page.
+* For instructions on how to get started with Data Prepper, see [Getting started with Data Prepper](https://opensearch.org/docs/2.6/data-prepper/getting-started/).
```

Comment:
Should we link to the latest Data Prepper getting started page? https://opensearch.org/docs/latest/data-prepper/getting-started/